src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/LinkDesignerBase.cs

```diff
@@ -69,8 +70,16 @@ protected bool FindResourceDesigner (AssemblyDefinition assembly, bool mainAppli
 		protected void ClearDesignerClass (TypeDefinition designer)
 		{
 			LogMessage ($"    TryRemoving {designer.FullName}");
-			designer.NestedTypes.Clear ();
-			designer.Methods.Clear ();
+			// for each of the nested types clear all but the
+			// int[] fields.
+			foreach (var nestedType in designer.NestedTypes) {
+				RemoveFieldsFromType (nestedType, designer.Module);
+				if (nestedType.Fields.Count == 0) {
+					// no fields we do not need any methods like the static ..cctor
+					nestedType.Methods.Clear ();
+				}
+			}
```

Comment:
Can we remove the entire type? something like:
```suggestion
			foreach (var nestedType in designer.NestedTypes.ToArray ()) { // or reverse for-loop to avoid LINQ
				RemoveFieldsFromType (nestedType, designer.Module);
				if (nestedType.Fields.Count == 0) {
					// no fields we do not need this type at all
					designer.NestedTypes.Remove (nestedType);
				}
			}
```

Or would this break something?